addie/processing/idl/run_sum_scans.py

```diff
@@ -94,21 +109,78 @@ def create_output_file(self):
+            list_element = chem_form_temp.split(" ")
+            formated_ele_list = []
+            for _element in list_element:
+                [formated_element, number_of_atoms, case] = get_periodictable_formatted_element_and_number_of_atoms(_element)
+                formated_ele_list.append(formated_element)
+            sample_form_str = ""
+            for item in formated_ele_list:
+                if case == 1:
+                    to_append = item + "_"
+                elif case == 2:
+                    to_append = re.findall('[A-Z][a-z]*', item)[-1]
+                    to_append += ("_" + re.findall('[0-9]+', item)[-1] + "_")
+                else:
+                    to_append = item.split("[")[1].split("]")[0] + item.split("[")[0] + "_" + item.split("]")[1] + "_"
+                sample_form_str += to_append
+            sample_form_str = sample_form_str[:-1]
+            f.write("{0:13s}{1:<s}\n".format("formula", sample_form_str))
+            f.write("{0:13s}{1:<s}\n".format("massdensity", self._mass_den[_label]))
+            f.write("{0:13s}{1:<s}\n".format("radius", self._radius[_label]))
+            f.write("{0:13s}{1:<s}\n".format("packfrac", self._pack_frac[_label]))
+            f.write("{0:13s}{1:<s}\n".format("geometry", self._geom[_label]))
+            [r_range_min, r_range_max] = o_gui_handler.get_r_range()
+            r_max_possible = "{0:6.2F}".format(np.pi / self.q_interval_min)
+            if r_range_min and r_range_max:
+                f.write("{0:13s}{1:<s}\n".format("rfilter", r_range_min + "," + r_range_max + "," + r_max_possible))
+            if q_range_min and q_range_max:
+                f.write("{0:13s}{1:<s}\n".format("qfilter", q_range_min + "," + q_range_max))
+            f.close()
+
     def collect_runs_checked(self):
         table = self.parent.table
+        _runs_name = {}
         _runs = {}
+        _sam_formula = {}
+        _mass_den = {}
+        _radius = {}
+        _pack_frac = {}
+        _geom = {}
         for _row_index in range(table.rowCount()):
             _selected_widget = table.cellWidget(_row_index, 0).children()[1]
             if (_selected_widget.checkState() == Qt.Checked):
-                _label = str(table.item(_row_index, 1).text())
-                _value = str(table.item(_row_index, 2).text())
-                _runs[_label] = _value
+                _runs = self.load_table(_runs, table, _row_index, 2)
+                # for Joerg's new script.
+                _runs_name = self.load_table(_runs_name, table, _row_index, 1)
+                _sam_formula = self.load_table(_sam_formula, table, _row_index, 3)
+                _mass_den = self.load_table(_mass_den, table, _row_index, 4)
+                _radius = self.load_table(_radius, table, _row_index, 5)
+                _pack_frac= self.load_table(_pack_frac, table, _row_index, 6)
+                _geom = self.load_table(_geom, table, _row_index, 7)
```

Comment:
So curious, why not add the name, formula, mass density, radius, packing fraction, and geometry as keys in the `_runs` dictionary, instead of returning now a list?

For the dictionary, the key lets you know some sort of metadata about the value in the dictionary.
For a list, the person has to know the index, which really contains no metadata for that entry.

Reply:
new codes pushed to address this issue